presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchRecordCursor.java

```diff
@@ -251,10 +255,45 @@ private Object getFieldValue(int field)
     private void extractFromSource(SearchHit hit)
     {
         Map<String, Object> map = hit.getSourceAsMap();
-        for (Map.Entry<String, Object> entry : map.entrySet()) {
-            String jsonPath = entry.getKey();
-            Object entryValue = entry.getValue();
-            setFieldIfExists(jsonPath, entryValue);
+        TreeMap<String, Object> fieldsMap = new TreeMap<>((left, right) -> {
+            int leftLength = left.split(FIELD_DELIMITER_REGEX).length;
+            int rightLength = right.split(FIELD_DELIMITER_REGEX).length;
+            if (leftLength == rightLength) {
+                return left.compareTo(right);
```

Comment:
The logic is a bit hard to follow because it interleaves string processing with nested field processing. The `compareTo` is a bit weird; for example `a.b` comparing with `ac.d` is going to compare `.` and `c`..
Is it possible to leverage `SubfieldTokenizer` and `Subfield` to separate string processing and nested field processing?